components/tikv_util/src/sys/disk.rs

```diff
@@ -82,15 +87,40 @@ pub fn get_disk_status(_store_id: u64) -> DiskUsage {
     }
 }
 
+/// Returns tuple of (total_space, available_space) for given path.
+///
+/// Attention, this function runs disk check in separate thread with 1 second
+/// timeout to avoid blocking.
 pub fn get_disk_space_stats<P: AsRef<Path>>(path: P) -> std::io::Result<(u64, u64)> {
-    fail_point!("mock_disk_space_stats", |stats| {
-        let stats = stats.unwrap();
-        let values = stats.split(',').collect::<Vec<_>>();
-        Ok((
-            values[0].parse::<u64>().unwrap(),
-            values[1].parse::<u64>().unwrap(),
-        ))
+    // 5 seconds timeout for disk space check, which is enough for most cases.
+    const DISK_STATS_CHECK_TIMEOUT: Duration = Duration::from_secs(5);
+
+    // Use a timeout channel to avoid blocking the thread when checking disk space.
+    let path = path.as_ref().to_owned();
+    let (tx, rx) = mpsc::channel();
+
+    // It's ok to use a separate thread to check disk space stats, because the
+    // access to disk is not frequent, and the disk check is not the hot path
+    // of tikv, so the overhead of allocating a thread is acceptable.
+    thread::spawn(move || {
```

Comment:
It would create and destroy threads too frequently, better to use a worker

Reply:
IMO, the frequency is low and acceptable, as this function is primarily called by:
1. StoreHeartbeat, which runs at a default interval of 10 seconds.
2. To check whether the disk is full or not, also at an interval of 10 seconds.